### PR TITLE
feat(parsed_transaction_history): add transaction pagination

### PR DESCRIPTION
**Pull request by rhh4x0r**
Added ability paginate the results of the request in the transaction history, by doing the following:

1. Creating a new request type called ParseTransactionHistoryRequest that has an `address` String and an Option `before` String
2. Updated `parsed_transaction_history` to input the new type, and update the API call if `before` exists. Updated documentation above it.
3. Updated the example to reflect the change

I've never contributed to an open-source project, so please forgive me if this is the wrong process for suggesting edits or updates to the module. 

I looked through the code to adhere to similar structures that y'all used in other aspects, but either way this should be a complete fix here or a good start to this implementation.

Any feedback or comments please let me know. Otherwise it should be good to go.

Cheers,

rhh

### CODE REVIEW

File: src/enhanced_transactions.rs [hunk 1]
```diff
@@ -31,14 +31,20 @@ impl Helius {
     ///
     /// # Arguments
     /// * `address` - An address for which a given parsed transaction history will be retrieved
```
Comment (0xIchigo) [inline src/enhanced_transactions.rs:33]: We'll need to update this to reflect the new request type as the `address` argument will no longer be valid

File: src/types/enhanced_transaction_types.rs [hunk 2]
```diff
@@ -210,6 +210,12 @@ pub struct ParseTransactionsRequest {
     pub transactions: Vec<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ParseTransactionHistoryRequest {
```
Comment (0xIchigo) [inline src/types/enhanced_transaction_types.rs:214]: I'd prefer if we renamed it to `ParsedTransactionHistoryRequest` so it follows the same name as its function with request added at the end

**Review by 0xIchigo — CHANGES_REQUESTED** [review]
Wow, thank you for your contribution! That was a fast turnaround, from raising the issue in Discord to making a PR! I've left a few minor comments, but this is a solid PR overall! 

Can you also please run the `cargo fmt` command before you commit your new changes? That way everything is formatted properly and all the checks for the build pass

File: src/enhanced_transactions.rs [hunk 3]
```diff
@@ -31,14 +31,20 @@ impl Helius {
...
+    /// * `before` - An optional signature to get history before, useful for pagination
```
Comment (0xIchigo) [inline src/enhanced_transactions.rs:34]: Can you rephrase this to the following, please:
```
* `before` - An optional signature that, when provided, fetches the parsed transaction history before the given signature. This is useful for pagination
```

File: src/enhanced_transactions.rs (hunk 1 above)
Comment (rhh4x0r) [reply src/enhanced_transactions.rs:33]: parse_transactions follows this format as well (showing the parameters of the type) which is why I framed it that way. Would you like me to update that one as welll?

Comment (rhh4x0r) [reply src/enhanced_transactions.rs:33]: As in, I put `address` and `before` instead of `ParsedTransactionHistoryRequest', like parse_transactions does above 
CleanShot 2024-05-14 at 08 11 14@2x
CleanShot 2024-05-14 at 08 11 14@2x
CleanShot 2024-05-14 at 08 11 14@2x

Comment (0xIchigo) [reply src/enhanced_transactions.rs:33]: Ah, yeah, that's an oversight on my end — `parse_transactions` should have `ParseTransactionsRequest` as its argument and the description should stay the same

**Review by 0xIchigo — APPROVED** [review]
LGTM! Thanks, and congrats on your first open-source contribution! 🔥
